exporter/prometheusremotewriteexporter/helper.go

```diff
@@ -215,21 +220,47 @@ func getPromMetricName(metric *otlp.Metric, ns string) string {
 	return sanitize(b.String())
 }
 
-// Simple helper function that takes the <Signature String - *TimeSeries> map
-// and creates a WriteRequest from the struct -- can move to the helper.go file
-func wrapTimeSeries(tsMap map[string]*prompb.TimeSeries) (*prompb.WriteRequest, error) {
+// Helper function that takes the <Signature String - *TimeSeries> map
+// and creates a batch of WriteRequest from the struct
+func wrapAndBatchTimeSeries(tsMap map[string]*prompb.TimeSeries) ([]*prompb.WriteRequest, error) {
 	if len(tsMap) == 0 {
 		return nil, errors.New("invalid tsMap: cannot be empty map")
 	}
+
+	var requests []*prompb.WriteRequest
 	var tsArray []prompb.TimeSeries
+	missingLastBatch := true
```

Comment:
nit: the variable name is misleading; you are not really missing last batch :-) 

Challenge for you, can you make few modifications to your code such that you don't need a new temporarily boolean variable to figure out if you have residual stuff in `tsArray`. 

Hint: using `tsArray`'s metadata should be enough. 

Keep in mind, this is a nit comment, don't spend too much time on this comment.

Reply:
Oh, haha, I can use tsArray's length anyways since it would have been resized to 0 if it was just processed and added into the batches. :")